cvmfs/util/algorithm.cc

```diff
@@ -69,6 +69,131 @@ double StopWatch::GetTime() const {
   return DiffTimeSeconds(start_, end_);
 }
 
+Log2Hist::Log2Hist(uint n)
+{
+  assert(n != 0);
+  this->number_of_bins = n;
+  this->bins = new atomic_int32[n + 1];  // +1 for overflow bin.
+  this->boundary_values = new uint[n + 1];  // +1 to avoid giant if statement
+  memset(this->bins, 0, sizeof(atomic_int32) * (n + 1));
+  memset(this->boundary_values, 0, sizeof(uint) * (n + 1));
+
+  uint i;
+  for (i = 1; i <= n; i++)
+  {
+    this->boundary_values[i] = (1 << ((i - 1) + 1));
+  }
+}
+
+Log2Hist::~Log2Hist()
+{
+  delete[] bins;
+  delete[] boundary_values;
+}
+
+atomic_int32* Log2Hist::GetBins()
+{
+  return this->bins;
+}
+
+string Log2Hist::Print()
+{
+  uint i = 0;
+
+  uint max_left_boundary_count = 1;
+  uint max_right_boundary_count = 1;
+  uint max_value_count = 1;
+  uint max_stars = 0;
+  uint max_bins = 0;
+  uint total_stars = 38;
+  uint total_sum_of_bins = 0;
+
+  for (i = 1; i <= number_of_bins; i++)
+  {
+    max_left_boundary_count = max(max_left_boundary_count,
```

Comment:
include algorithm (the C++ standard algorithm header) for `std::max`

Reply:
Done!